waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/client/CloseableThriftHiveMetastoreIfaceClientFactoryTest.java

```diff
@@ -15,82 +15,61 @@
+    when(federatedMetaStore.getRemoteMetaStoreUris()).thenReturn(THRIFT_URI);
+    factory = new CloseableThriftHiveMetastoreIfaceClientFactory();
   }
 
   @Test
   public void hiveConf() throws Exception {
-    ArgumentCaptor<HiveConf> hiveConfCaptor = ArgumentCaptor.forClass(HiveConf.class);
-
-    factory.newInstance(newFederatedInstance("fed1", THRIFT_URI));
-    verify(metaStoreClientFactory).newInstance(hiveConfCaptor.capture(), anyString(), anyInt());
-
-    HiveConf hiveConf = hiveConfCaptor.getValue();
-    assertThat(hiveConf.getVar(ConfVars.METASTOREURIS), is(THRIFT_URI));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_LOCALHOST.varname), is(nullValue()));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_PORT.varname), is(nullValue()));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_ROUTE.varname), is(nullValue()));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_KNOWN_HOSTS.varname), is(nullValue()));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_PRIVATE_KEYS.varname), is(nullValue()));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_SESSION_TIMEOUT.varname), is(nullValue()));
+    MetastoreClientFactoryHelper helper = new MetastoreClientFactoryHelper(federatedMetaStore);
+    CloseableThriftHiveMetastoreIface result = factory.newInstance(federatedMetaStore, helper);
+    assertThat(result, instanceOf(Proxy.class));
   }
 
   @Test
   public void hiveConfForTunneling() throws Exception {
-    ArgumentCaptor<HiveConf> hiveConfCaptor = ArgumentCaptor.forClass(HiveConf.class);
-
-    MetastoreTunnel metastoreTunnel = new MetastoreTunnel();
-    metastoreTunnel.setLocalhost("local-machine");
-    metastoreTunnel.setPort(2222);
-    metastoreTunnel.setRoute("a -> b -> c");
-    metastoreTunnel.setKnownHosts("knownHosts");
-    metastoreTunnel.setPrivateKeys("privateKeys");
-    metastoreTunnel.setTimeout(123);
-    AbstractMetaStore federatedMetaStore = newFederatedInstance("fed1", THRIFT_URI);
-    federatedMetaStore.setMetastoreTunnel(metastoreTunnel);
-
-    factory.newInstance(federatedMetaStore);
-    verify(metaStoreClientFactory).newInstance(hiveConfCaptor.capture(), anyString(), anyInt());
-
-    HiveConf hiveConf = hiveConfCaptor.getValue();
-    assertThat(hiveConf.getVar(ConfVars.METASTOREURIS), is(THRIFT_URI));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_LOCALHOST.varname), is("local-machine"));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_PORT.varname), is("2222"));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_ROUTE.varname), is("a -> b -> c"));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_KNOWN_HOSTS.varname), is("knownHosts"));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_PRIVATE_KEYS.varname), is("privateKeys"));
-    assertThat(hiveConf.get(WaggleDanceHiveConfVars.SSH_SESSION_TIMEOUT.varname), is("123"));
+    when(helper.get()).thenReturn(tunnelingMetaStoreClientFactory);
```

Comment:
The class doesn't know about tunnels anymore nor should the test, you are just testing mocks

Reply:
Should I just remove this test then?